### ISSUE TITLE
unit tests

### ISSUE THREAD

**Issue by fahad-o** [cbraswe/video#13]
Add more unit tests. The unit tests are methods that have an assertion (e.g., self.assertEqual(some_property, something)). I know one of the methods does not have an assertion, and that is ok because the idea from it is to check if the program runs).

By the way, before adding any new modification, please run the vectorizer file, and after the modification, run the unit test file to see if it runs on any output dimension and if it is faster than the old version's time (try not to use any program).

**Comment by cbraswe** [comment]
Partially related, but I got the question from looking at the unit test file.

For metrics and the report, do you see value in running different output dimensions and capturing the results? 

I think I do because parts of the code will perform differently based on the output dimensions. If the output width is smaller, there's probably more average calculations and less np.repeat() towards the end. So, some iterations of our code will perform well with large output dimensions and others may perform well with small output dimensions.

**Comment by fahad-o** [comment]
> Partially related, but I got the question from looking at the unit test file.
> 
> For metrics and the report, do you see value in running different output dimensions and capturing the results?
> 
> I think I do because parts of the code will perform differently based on the output dimensions. If the output width is smaller, there's probably more average calculations and less np.repeat() towards the end. So, some iterations of our code will perform well with large output dimensions and others may perform well with small output dimensions.

definitely, there is value and I think someone should be constantly evaluating changes so we don't miss our stop on the road for optimization.
